packages/functional-tests/tests/syncV3/signIn.spec.ts

```diff
@@ -0,0 +1,190 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { test, expect } from '../../lib/fixtures/standard';
+import { EmailHeader, EmailType } from '../../lib/email';
+
+const password = 'passwordzxcv';
+let email;
+
+test.describe('Firefox Desktop Sync v3 sign in', () => {
+  test.beforeEach(async ({ pages: { login } }) => {
+    test.slow();
+    email = login.createEmail('sync{id}');
+    await login.clearCache();
+  });
+
+  test.afterEach(async ({ pages: { login } }) => {
+    await login.clearCache();
+  });
+
+  test('verified, does not need to confirm ', async ({
+    target,
+    page,
+    pages: {
+      login,
+      signinTokenCode,
+      connectAnotherDevice,
+      fxDesktopV3ForceAuth,
+    },
+  }) => {
+    const uaStrings = {
+      desktop_firefox_58:
+        'Mozilla/5.0 (Windows NT 6.1; WOW64; rv:58.0) Gecko/20100101 Firefox/58.0',
+    };
```

Comment:
Looking at the 5 tests, we could refactor some of the stuff into a helper (not necessarily in this PR, but at a future date).

Rough, 100% untested idea:

```js
async function setupTest(options = {}) {
  // Not sure if this belongs here, or at top of file. It's only used in one place, so I'm indifferent.
  const uaStrings = {
    desktop_firefox_58: 'Mozilla/5.0 (Windows NT 6.1; WOW64; rv:58.0) Gecko/20100101 Firefox/58.0',
  };
  // `context` and `service` seem to be the same for each test, but sometimes we add `action=email` as well.
  let query = {
    forceUA: uaStrings.desktop_firefox_58,
    context: 'fx_desktop_v3',
    service: 'sync',
  };
  // This allows us to customize the query string by passing in an optional `options.query` object.
  // Will be a minor tweak if we want to pass `action=email` style string instead of `{action:'email'}` object.
  if (options.query) {
    query = Object.assign(query, options.query);
  }
  // Convert object to query string.
  const queryParam = new URLSearchParams(query);
  // Support for optional stuff like `'blocked{id}'` prefix.
  const email = login.createEmail(options.prefix);
  // Not all tests did this `target.auth.signUp()` step, so another `if` statement.
  if (options.doSignup) {
    // Optional support for passing a config object into the `.signUp()` method (stuff like `lang` and `preVerified`).
    await target.auth.signUp(email, password, options.cfg || {});
  }
  await page.goto(`${target.contentServerUrl}?${queryParam.toString()}`);
  // ... more here? The test start diverging a bit at this point.
  return email;
}
```

```js
const email = setupTest({
  query: {
    action: 'email'
  },
  prefix: 'blocked{id}',
  cfg: {
    lang: 'en',
    preVerified: true
  }
});
```